test/helpers/governance.js

```diff
@@ -196,6 +197,44 @@ class GovernorHelper {
   }
 }
 
+/**
+ * Encodes a list ProposalStates into a bytes32 representation where each bit enabled corresponds to
+ * the underlying position in the `ProposalState` enum. For example:
+ *
+ * 0x000...10000
+ *   ^^^^^^------ ...
+ *         ^----- Succeeded
+ *          ^---- Defeated
+ *           ^--- Canceled
+ *            ^-- Active
+ *             ^- Pending
+ */
+function proposalStatesToBitMap(proposalStates, options = {}) {
+  if (!Array.isArray(proposalStates)) {
+    proposalStates = [proposalStates];
+  }
+  const statesRange = Object.keys(ProposalState).length - 1;
+  let result = 0;
+
+  const uniqueProposalStates = new Set(proposalStates.map(bn => bn.toNumber())); // Remove duplicates
+  for (const state of Array.from(uniqueProposalStates)) {
+    if (state < 0 || state > statesRange) {
+      expect.fail(`ProposalState ${state} out of possible states (0 - ${statesRange})`);
+    } else {
+      result |= 1 << state;
+    }
+  }
+
+  if (options.inverted) {
+    const mask = 2 ** statesRange - 1;
```

Comment:
I think there is an error in this function, the inverted mask in line 229 is smaller than it should be. This should fix it. Including a few other minor suggestions as well...

```suggestion
  const statesCount = Object.keys(ProposalState).length;
  let result = 0;

  const uniqueProposalStates = new Set(proposalStates.map(bn => bn.toNumber())); // Remove duplicates
  for (const state of Array.from(uniqueProposalStates)) {
    if (state < 0 || state >= statesCount) {
      expect.fail(`ProposalState ${state} out of possible states (0...${statesCount}-1)`);
    } else {
      result |= 1 << state;
    }
  }

  if (options.inverted) {
    const mask = 2 ** statesCount - 1;
```